gateway-ha/src/main/java/io/trino/gateway/ha/router/RoutingManager.java

```diff
@@ -185,6 +225,12 @@ protected String findBackendForUnknownQueryId(String queryId)
         return gatewayBackendManager.getActiveAdhocBackends().get(0).getProxyTo();
     }
 
+    protected String lookupBackendForCookie(String cookie)
+    {
+        Optional<String> backend = cacheManager.getBackendForCookie(cookie);
+        return backend.orElseGet(() -> provideAdhocBackend(""));
+    }
+
     // Predicate helper function to remove the backends from the list
     // We are returning the unhealthy (not healthy)
     private boolean isBackendNotHealthy(String backendId)
```

Comment:
Sidenode .. this would be better as isBackendHealthy ..

Reply:
@andythsu ☝️